can_tools/scrapers/puppet.py

```
@@ -174,14 +203,23 @@ async def _get(self) -> pd.DataFrame:
                 button = await page.waitForSelector(self.download_button_id)
                 await button.click()
 
-                # find and click crosstab option to initiate download
-                crosstab = await page.waitForXPath(
-                    f"//button[{self.crosstab_button_selector}]"
-                )
-                await crosstab.click()
-
-                # TODO: better way to wait for download?
-                await asyncio.sleep(5)
+                for click_selector in self.tableau_click_selector_list:
+                    if hasattr(click_selector, "__call__"):
+                        # Use for special handling in an instance specific function
+                        await click_selector(page, tmpdirname)
+                    else:
+                        selector = await page.waitForXPath(f"//*[{click_selector}]")
+                        await selector.click()
+                        await asyncio.sleep(2)
+
+                # Wait for download to finish. 10 cycles of 2 seconds is arbitrary
+                for _ in range(10):
+                    if len(os.listdir(tmpdirname)) > 0:
+                        break
+                    await asyncio.sleep(2)
+                else:
+                    # TODO0: ZW: Throw / report error?
```

Yes, let's throw here as something unexpected happened

---

Added assert